js/app.js

```diff
@@ -452,6 +528,22 @@
 	*/
 
 	// implement function here
+	const spliceItems = (items, i, list) => {
+		for (let i = 0; i < items.length; i++) {
+			if (items[i].hasOwnProperty('item') && items[i].hasOwnProperty('price')) {
+				list.splice(i, 0, items[i])
+        if (i > list.length){
+          list.push(items[i]);
```

Comment:
It doesn't look like this is doing what you want :( But that's ok, it's just one thing out of the whole suite.